velox/duckdb/conversion/DuckParser.cpp

```diff
@@ -358,8 +358,21 @@ std::shared_ptr<const core::IExpr> parseOperatorExpr(
 
     TypePtr valueType = UNKNOWN();
     for (auto i = 0; i < numValues; i++) {
-      if (auto constantExpr = dynamic_cast<ConstantExpression*>(
-              operExpr.children[i + 1].get())) {
+      auto valueExpr = operExpr.children[i + 1].get();
+      if (const auto castExpr = dynamic_cast<CastExpression*>(valueExpr)) {
+        if (castExpr->child->GetExpressionType() ==
+            ExpressionType::VALUE_CONSTANT) {
+          auto constExpr =
+              dynamic_cast<ConstantExpression*>(castExpr->child.get());
+          auto value =
```

Comment:
Please check constExpr is non-null before de-referencing it in the next line. We should ensure that cast is not over any other expression here.